pkg/cli/build.go

```diff
@@ -0,0 +1,825 @@
+package cli
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	bk_io "github.com/buildkite/cli/v3/internal/io"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	buildkite "github.com/buildkite/go-buildkite/v4"
+	"github.com/pkg/browser"
+	"gopkg.in/yaml.v3"
+)
+
+// Build commands
+type BuildCmd struct {
+	Cancel   BuildCancelCmd   `cmd:"" help:"Cancel a build"`
+	Download BuildDownloadCmd `cmd:"" help:"Download build artifacts or job logs"`
+	New      BuildNewCmd      `cmd:"" help:"Create a build"`
+	Rebuild  BuildRebuildCmd  `cmd:"" help:"Rebuild a build"`
+	View     BuildViewCmd     `cmd:"" help:"View a build"`
+	Watch    BuildWatchCmd    `cmd:"" help:"Watch build status until completion"`
+}
+
+type BuildCancelCmd struct {
+	Build string `arg:"" help:"Build number (pipeline auto-detected from git repo) or build URL"`
+}
+
+type BuildDownloadCmd struct {
+	Build       string `arg:"" help:"Build number (pipeline auto-detected from git repo) or build URL"`
+	Destination string `help:"Download destination directory"`
+	Job         string `help:"Job ID to filter artifacts"`
+	Pattern     string `help:"Artifact name pattern"`
+	Logs        bool   `help:"Download job logs instead of artifacts"`
+	FailedOnly  bool   `help:"Only download logs from failed jobs (requires --logs)" name:"failed-only"`
```

Comment:
Do we emit error/warning when `--failed-only` is passed without `--logs`, or fail silently?

Reply:
we error now, good catch!